lib/blockchain.go

```diff
@@ -2393,14 +2385,20 @@ func (bc *Blockchain) CreatePrivateMessageTxn(
 		PublicKey: senderPublicKey,
 		TxnMeta: &PrivateMessageMetadata{
 			RecipientPublicKey: recipientPublicKey,
-			EncryptedText:      encryptedText,
+			EncryptedText:      encryptedMessageHex,
 			TimestampNanos:     tstampNanos,
 		},
 
 		// We wait to compute the signature until we've added all the
 		// inputs and change.
 	}
 
+	// Add {V2 : 1} field to ExtraData to indicate we are encrypting
+	// using shared secret.
+	messageExtraData := make(map[string][]byte)
+	messageExtraData["V2"] = UintToBuf(1)
```

Comment:
Maybe instead we name this "V" and set the value to "2". Feels more future proof.